tools/publish-kubebuilder-tools/main.go

```diff
@@ -0,0 +1,445 @@
+package main
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"io/fs"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"cloud.google.com/go/storage"
+)
+
+func main() {
+	pullSecretFile := flag.String("pull-secret", "", "The pull secret to use for the kubebuilder tools")
+	version := flag.String("version", "", "The version of the kubebuilder tools to publish. This should be a Kubernetes version that the build is based upon.")
+	outputDir := flag.String("output-dir", "", "The output directory to write the kubebuilder tools to")
+	payload := flag.String("payload", "", "The payload to use for building the kubebuilder tools archives. This should be in the format registry.ci.openshift.org/ocp/release:<version>")
+	skipUpload := flag.Bool("skip-upload", false, "Skip uploading the artifacts created to the openshift-gce-devel/openshift-kubebuilder-tools bucket")
+
+	flag.Parse()
+
+	if *pullSecretFile == "" {
+		panic("pull-secret is required")
```

Comment:
Give me a hint for how to create this.  Do I log into a CI cluster and do the `oc registry login` thing?

Reply:
Added a note to the makefile where this is being called